### ISSUE TITLE
default option: start client at boot

### ISSUE THREAD

**Issue by Erkan-Yilmaz** [Erkan-Yilmaz/Gridcoin-tasks#20]
reason: 
* participation in staking

todo: 
* I've no windows, so I am asking around people

mentioned in [hangout 2](https://github.com/Erkan-Yilmaz/Gridcoin-hangout-minutes/blob/master/hangout_2016_08_13.MD): 27:40

**Comment by grctest** [comment]
I believe that https://github.com/Scalextrix has been working on auto-startup scripts/methods on raspberry pi's.

**Comment by Scalextrix** [comment]
Yes I have created the gridcoinresearchd as a service on a raspberry pi, this should work on any Linux distro that uses systemd. 
I will post some instructions tomorrow

**Comment by TG13088** [comment]
I run Win 7 x64 Pro and even though the check-box in Settings is marked, I still have to start the Wallet manually.  This is not an issue for me since I prefer having some measure of control over what executes at Start-up ... particularly when I am performing maintenance or trouble-shooting.  I would recommend against Auto-Start as being the default for such reasons.  A desktop icon with the appropriate properties should be good enough ... not that hard to remember to double-click an icon in Windows ;-)

**Comment by Scalextrix** [comment]
The following will work on Linux that has systemd:
pi@raspberrypi:~ $ gridcoinresearchd stop

pi@raspberrypi:~ $ cd /etc/systemd/system

pi@raspberrypi:~ $ sudo nano gridcoinresearchd.service

Enter the following

[Unit]
Description=Gridcoin daemon services
After=tlp-init.service

[Service]
Type=simple
RemainAfterExit=yes
ExecStart=/usr/bin/gridcoinresearchd   
Restart=always
RestartSec=60
User=pi

[Install]
WantedBy=multi-user.target


CTRL+X, Y and Enter

pi@raspberrypi:~ $ systemctl enable gridcoinresearchd.service

pi@raspberrypi:~ $ systemctl start gridcoinresearchd.service

pi@raspberrypi:~ $ systemctl status gridcoinresearchd.service

Should output something like:

? gridcoinresearchd.service - Gridcoin daemon services
   Loaded: loaded (/lib/systemd/system/gridcoinresearchd.service; enabled)
   Active: active (exited) since Sat 2016-12-10 16:22:58 GMT; 20h ago
  Process: 409 ExecStart=/usr/bin/gridcoinresearchd (code=exited, status=0/SUCCESS)
 Main PID: 409 (code=exited, status=0/SUCCESS)
   CGroup: /system.slice/gridcoinresearchd.service
           +-613 /usr/bin/gridcoinresearchd

In top you should see gridcoinresearchd running with the process id in the last line (mine is 613)

**Comment by TheCharlatan** [comment]
The startup tick works fine with my linux system (opensuse 42.1)

**Comment by MagixInTheAir** [comment]
Here are my 2 cents : 
- The wallet should autostart at boot (with delay ?) if not encrypted or auto-unlock is setup
- It should be possible to setup auto-unlock through the wallet, and at the end of the wizard autostart should be proposed as an option.
Thought ?

**Comment by Scalextrix** [comment]
I guess auto unlock for staking could really be a checkbox, it's only generating a hash and auto adding it to gridcoinresearch.conf, someone smart can do that in the core wallet Im sure.

I added a delayed start for the Qt wallet in Windows Task Scheduler, again someone with enough smarts could probably automate that.

**Comment by tomasbrod** [comment]
What is the point of encrypting your wallet if it unlocks automatically?

**Comment by Scalextrix** [comment]
It only unlocks for staking, so you still have to fully unlock it to make any other transactions

**Comment by tomasbrod** [comment]
Don't be fool. The "for staking only" is just a flag, that prevents signing non-PoS transactions. The wallet is basically unlocked.

**Comment by MagixInTheAir** [comment]
This has nothing to do with autostart, this is a general security problem the users should be made aware of... but doesn't change the autostart thing. If the user accepts auto-unlock (with its problems) and autostart, then I don't see a problem as long as everything is made clear

**Comment by Scalextrix** [comment]
Are we saying unlocking the wallet for staking is insecure?
